### PR TITLE
[spv-out] Clean up capability handling.

### PR DESCRIPTION
**Pull request by jimblandy**
Remove 'forbidden_caps'.

Accumulate capabilities actually used separately from the permitted
capabilities, so that the latter can be retained across Writer resets, while the
former is cleared between modules.

### CODE REVIEW

File: src/back/spv/mod.rs [hunk 1]
```diff
@@ -385,8 +385,18 @@ pub struct Writer {
...
-    forbidden_caps: Option<&'static [Capability]>,
+
+    /// The set of capabilities modules are permitted to use.
+    ///
+    /// This is initialized from `Options::capabilities`.
+    capabilities_available: Option<crate::FastHashSet<Capability>>,
+
+    /// The set of capabilities used by this module.
+    ///
+    /// If `capabilities_available` is `Some`, then this is always a subset of
```
Comment (kvark) [inline src/back/spv/mod.rs:396]: what's the motivation for collecting this subset if `capabilities_available` is provided?

**Review by kvark — COMMENTED** [review]
Looks very nice, just one concern here

File: src/back/spv/mod.rs (hunk 1 above)
Comment (jimblandy) [reply src/back/spv/mod.rs:396]: We use it to emit `OpCapability` instructions.

Comment (kvark) [reply src/back/spv/mod.rs:396]: I don't think it helps in any way versus just passing `capabilities_available` (the superset).

**Review by kvark — APPROVED** [review]

**Comment by kvark** [comment]
Published in 0.6.2

**Pull request merged**